schemas/donor.json

```diff
@@ -0,0 +1,59 @@
+{
+  "description": "The collection of data elements related to a specific donor in a specific ICGC-ARGO program.",
+  "fields": [
+    {
+      "description": "Unique identifier of the ARGO program.",
+      "name": "program_id",
+      "restrictions": {
+        "required": true
+      },
+      "valueType": "string"
+    },
+    {
+      "description": "Unique identifier of the donor, assigned by the data provider.",
+      "name": "submitter_donor_id",
+      "restrictions": {
+        "required": true
+      },
+      "valueType": "string"
+    },
+    {
+      "description": "Donors last known state of living or deceased.",
+      "name": "vital_status",
+      "restrictions": {
+        "codeList": [
+          "Alive",
+          "Deceased",
+          "Unknown"
```

Comment:
The optional fields can be filled in by a default - i forget the method to indicate that.  I will have to talk to the devs for how to format that. 

I understand `Not Reported` as the question was asked, but the patient chose not to answer and `Unknown` as we were not able to get that information for some reason. In the case that we assign a default value when a user does not submit, i feel like `Unknown` in this definition would be more accurate.  Is this understanding correct @hknahal ?

Reply:
`Not reported` was removed from the `central_pathology_confirmed` field, so there is only `Yes`, `No` and `Unknown`.

`Unknown`: means the clinical data is missing, and therefore the field is considered _incomplete._
`Not applicable` mean the clinical field is not applicable to the cancer type, but the clinical field is _not considered incomplete._

This differentiation is important for clinical completeness metrics where a field that is missing (unknown) is counted differently than a field that is not applicable.

We shouldn't always include `Unknown` or `Not applicable` is all codelists because some clinical fields are mandatory and we require data for these fields. They can however be included in codelists for optional clinical fields.

The default value for optional fields can be `Not applicable`, which means the data submitter did not submit clinical data for the optional field because it does not apply to the cancer type.

Reply:
> The default value for optional fields can be `Not applicable`, which means the data submitter did not submit clinical data for the optional field because it does not apply to the cancer type.

Not sure about defaulting to `Not applicable` for optional fields when not submitted. It should be safer to let submitter to explicit mark a field `Not applicable`, because it could be wrong.

Reply:
@junjun-zhang Can you give an example of when it could be wrong to use `Not applicable` as default for an optional field?

Reply:
It seems we don't have optional field in this PR, but looking at the Extended Clinical fields, things like `height`, `weight`, `bmi` are optional. If a submitter does not submit them, can it be interpreted as `Not applicable`? What if the submitter just does not have this information at the time of submission? In such scenario, `Unknown` should be more suitable.

Reply:
Maybe a better term for default is neither, but instead for the default value to be `Not submitted`.  In this case, the valuable semantics of `Unknown` and `Not applicable` can be maintained for clinical completeness, but with the addition of Not Submitted we can accurately count clinical completeness.  @hknahal what do you think?

Reply:
I agree with @rosibaj - `Not submitted` can be used as the default for optional fields that apply to any cancer type. We tend to get a lot of questions for ICGC clinical data that is missing, and it's simply because the data was not submitted to us. The optional clinical fields `bmi`, `height` and `weight` will be accepting `integer` values, so we'll need a way to denote this (ie. "Unknown", "Not submitted).

For optional fields which are cancer-specific, the default term `Not applicable` can be used. For example, there will be a pancreatic cancer specific field called `margin_pancreatic_resection_margin`. If a lung cancer project is submitting data, this field does not apply, so the default value for `margin_pancreatic_resection_margin` would be `Not applicable`.

Reply:
indeed, we need an unambiguous way to denote `Not submitted`. This **could** work: the absence of a field or its value in submitted files denotes `Not submitted`. In TSV, either the field is entirely missing or when field exists but the value in a particular row is empty string (unless we needs empty string to denote any other semantics).

On the other note, how to define a field is only applicable to a subset of cancers is something probably to be handled by script based validation.

Reply:
I've discussed with Jon and Dusan - before we go any further i would like to understand the requirements for tracking not-submitted further!